testdata/p4_16_samples_outputs/issue1937-1-bmv2-frontend.p4

```diff
@@ -0,0 +1,60 @@
+#include <core.p4>
+#include <v1model.p4>
+
+header h1_t {
+    bit<8> f1;
+    bit<8> f2;
+}
+
+struct headers_t {
+    h1_t h1;
+}
+
+struct metadata_t {
+}
+
+control ingressImpl(inout headers_t hdr, inout metadata_t meta, inout standard_metadata_t stdmeta) {
+    @name(".foo") action foo(out bit<8> x, in bit<8> y=5) {
+        x = y >> 2;
+    }
+    @name(".foo") action foo_0(out bit<8> x_1, in bit<8> y_1=5) {
```

Comment:
Having two actions with `@name(".foo")` annotations looks like wrong to me.  If the user did this in their original source program, the compiler does not catch it as an error?

Reply:
Does it make sense that the compiler is taking one action, and creating two separate actions from it?  Why is that useful?

Reply:
The code makes sure that each action is used in a single table. For some targets there is no action reuse anyway; if you have the same code you have to implement it twice. 
This also opens the opportunity to specialize the action body for each call site. Perhaps this should be a controllable option - whether to duplicate or not. But it's not wrong, in the sense that it preserves the program semantics.

Reply:
This is not an error, you can have multiple actions with the same name; the control-plane disambiguates them through the tables that invoke them.

Reply:
These are both invoked by the same table, so no such through-the-table disambiguation is possible here, is it?